pkg/engine/task/task_toggle.go

```diff
@@ -0,0 +1,65 @@
+package task
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1beta1"
+)
+
+const (
+	toggleTaskError = "ToggleTaskError"
+)
+
+// ToggleTask will apply or delete a set of given resources to the cluster based on value of Parameter. See Run method for more details.
+type ToggleTask struct {
+	Name      string
+	Parameter string
+	Resources []string
+}
+
+func (tt ToggleTask) Run(ctx Context) (bool, error) {
+	// 1. - Get the task to run
+	task, err := tt.intermediateTask(ctx)
+	if err != nil {
+		return false, fatalExecutionError(err, toggleTaskError, ctx.Meta)
+	}
+	// 2. - Run the returned task
+	return task.Run(ctx)
+}
+
+func (tt ToggleTask) intermediateTaskSpec() v1beta1.TaskSpec {
+	return v1beta1.TaskSpec{
+		ResourceTaskSpec: v1beta1.ResourceTaskSpec{
+			Resources: tt.Resources,
+		},
+	}
+}
+
+func (tt ToggleTask) intermediateTask(ctx Context) (Tasker, error) {
+	var task Tasker
+	// 1. - Get the parameter value
+	val, exists := ctx.Parameters[tt.Parameter]
+	if !exists {
+		return task, fmt.Errorf("no value for parameter [%s] found", tt.Parameter)
+	}
+	enabled, err := strconv.ParseBool(val)
+	if err != nil {
+		return task, err
```

Comment:
Nit: Let's provide more context here
```suggestion
		return task, fmt.Errorf("could not parse value of parameter %s: %v", err)
```